### ISSUE TITLE
Use reliable QoS for publishers

### ISSUE THREAD

**Issue by eborghi10** [iRobotEducation/create3_sim#135]
We should apply a change that is already being used in `gazebo_ros_pkgs`, see [here](https://github.com/ros-simulation/gazebo_ros_pkgs/pull/1224).

I used this approach to connect with [`rosbridge_server`](https://github.com/RobotWebTools/rosbridge_suite/tree/ros2) for example and with the current configuration, it doesn't work.

```
[joint_state_publisher-3] /opt/ros/galactic/lib/python3.8/site-packages/rclpy/qos.py:307: UserWarning: DurabilityPolicy.RMW_QOS_POLICY_DURABILITY_TRANSIENT_LOCAL is deprecated. Use DurabilityPolicy.TRANSIENT_LOCAL instead.
```

**Comment by rjcausarano** [comment]
@eborghi10 Do you know of any issues that this change will fix? Do we run the risk of introducing new issues?

**Comment by rjcausarano** [comment]
This might fix https://github.com/iRobotEducation/create3_sim/issues/125

**Comment by eborghi10** [comment]
> @eborghi10 Do you know of any issues that this change will fix? Do we run the risk of introducing new issues?

On the opposite, I think it'll allow us to use RViz and another tooling.

**Comment by eborghi10** [comment]
> This might fix #125

@Ednip7 Let's double-check this issue too once you have the changes.
